### ISSUE TITLE
Marker is not rendered when using custom icon

### ISSUE THREAD

**Issue by Powerhamster** [netzmacht/php-leaflet#3]
When creating the icon for the marken an exception is throw
```
TypeError: this.callInitHooks is not a function
```
It can be solved by using `new` when creating the icon.

```
public function encodeIcon(ImageIcon $icon, Encoder $encoder)
    {
        return sprintf(
            '%s = new L.Icon(%s);',
            $encoder->encodeReference($icon),
            $encoder->encodeArguments(array($icon->getOptions()))
        );
    }
```

**Comment by dmolineus** [comment]
Yeah, it should use the factory methods Leaflet provides `L.icon` and `L.divIcon`. Never stumpled upon it because of using a custom implementation for custom icons.

**Comment by dmolineus** [comment]
Fixed in Version 1.1.0.
